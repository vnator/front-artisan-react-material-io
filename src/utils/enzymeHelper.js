import { IntlProvider, createIntl } from 'react-intl';
import { mount, shallow } from 'enzyme';
import { flattenMessages } from '../config/flattenMessages';
import { messages } from '../messages';

// You can pass your messages to the IntlProvider. Optional: remove if unneeded.

const defaultLocale = 'pt-BR';
const locale = defaultLocale;
const wrapMessages = flattenMessages(messages[locale]);

function mountWithIntl(node) {
  return mount(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

function shallowWithIntl(node) {
  return shallow(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

const intl = createIntl({
  locale: 'pt-BR',
  defaultLocale: 'pt-BR',
  messages: wrapMessages,
});

export { mountWithIntl, shallowWithIntl, intl };
