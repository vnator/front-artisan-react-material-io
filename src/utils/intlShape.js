import Type from 'prop-types';

const intlConfig = {
  locale: Type.string,
  timeZone: Type.string,
  messages: Type.objectOf(Type.string),
  defaultLocale: Type.string,
};

const intlFormatters = {
  formatDate: Type.func,
  formatTime: Type.func,
  formatRelativeTime: Type.func,
  formatNumber: Type.func,
  formatPlural: Type.func,
  formatMessage: Type.func,
  formatHTMLMessage: Type.func,
};

const intlShape = {
  ...intlConfig,
  ...intlFormatters,
};

export { intlShape, intlConfig, intlFormatters };
