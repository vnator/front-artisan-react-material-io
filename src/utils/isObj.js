const isObj = obj => typeof obj === 'object' && !(obj instanceof Array);

export { isObj };
