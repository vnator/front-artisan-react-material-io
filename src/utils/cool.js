import { isObj } from './isObj';

const cool = (a) =>
  Object.freeze(
    Object.entries(a).reduce(
      (total, [key, val]) =>
        isObj(val)
          ? Object.freeze({ ...total, [key]: cool(val) })
          : Object.freeze({ ...total, [key]: val, isFrozen: true }),
      Object.freeze(),
    ),
  );

export { cool };
