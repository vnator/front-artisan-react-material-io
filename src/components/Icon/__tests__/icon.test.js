import React from 'react';
import { IconFlatDown } from '../IconFlatDown';
import { IconFlatLeft } from '../IconFlatLeft';
import { IconFlatRight } from '../IconFlatRight';
import { IconFlatUp } from '../IconFlatUp';
import { IconReact } from '../IconReact';
import { Svg } from '../Svg';
import Enzyme from 'enzyme';
import { mountWithIntl } from '../../../utils/enzymeHelper';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Icons', () => {
  it('Snapshot IconFlatDown', () => {
    const icon = mountWithIntl(<IconFlatDown />);

    expect(icon).toMatchSnapshot();
  });

  it('Snapshot IconFlatLeft', () => {
    const icon = mountWithIntl(<IconFlatLeft />);

    expect(icon).toMatchSnapshot();
  });

  it('Snapshot IconFlatRight', () => {
    const icon = mountWithIntl(<IconFlatRight />);

    expect(icon).toMatchSnapshot();
  });

  it('Snapshot IconFlatUp', () => {
    const icon = mountWithIntl(<IconFlatUp />);

    expect(icon).toMatchSnapshot();
  });

  it('Snapshot IconReact', () => {
    const icon = mountWithIntl(<IconReact />);

    expect(icon).toMatchSnapshot();
  });

  it('Snapshot Svg', () => {
    const icon = mountWithIntl(
      <Svg>
        <IconReact />
      </Svg>,
    );

    expect(icon).toMatchSnapshot();
  });
});
