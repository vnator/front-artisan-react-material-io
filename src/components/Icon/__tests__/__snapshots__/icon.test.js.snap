// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Snapshot IconFlatDown 1`] = `ReactWrapper {}`;

exports[`Icons Snapshot IconFlatLeft 1`] = `ReactWrapper {}`;

exports[`Icons Snapshot IconFlatRight 1`] = `ReactWrapper {}`;

exports[`Icons Snapshot IconFlatUp 1`] = `ReactWrapper {}`;

exports[`Icons Snapshot IconReact 1`] = `ReactWrapper {}`;

exports[`Icons Snapshot Svg 1`] = `ReactWrapper {}`;
